src/torrent.cpp

```diff
@@ -61,7 +61,16 @@ POSSIBILITY OF SUCH DAMAGE.
 #include <cstdio> // for snprintf
 #include <functional>
 
-#include "libtorrent/torrent.hpp"
+#include "libtorrent/aux_/torrent.hpp"
+
+#ifdef TORRENT_SSL_PEERS
+#include "libtorrent/ssl_stream.hpp"
+#include "libtorrent/aux_/disable_warnings_push.hpp"
+#include <boost/asio/ssl/context.hpp>
+#include <boost/asio/ssl/verify_context.hpp>
```

Comment:
yes, these asio SSL headers are only for openssl, not gnutls. These are abstracted by ssl headers in libtorrent

Reply:
I see, I will review it and put the necessary `#if`

Reply:
I think this is what you want: https://github.com/arvidn/libtorrent/blob/master/include/libtorrent/ssl.hpp